### PR TITLE
Clear plots if processing fails in preview tab

### PR DESCRIPTION
**Pull request by robertapplin**
**Description of work.**
This PR ensures the reduction plot and region selector plot gets cleared when there is an error.

Two things we noted while working on this:
- The cursor info doesn't get cleared if you move your cursor out of the region selector plot fast
- The clearing of the reduction plot does not clear the axis limits or remove the legend handle

We thought we'd let the reviewer decide if these need to be fixed.

**To test:**
1. Open `ISIS Reflectometry` interface
2. Go to the Preview tab and load `INTER45455_inst`
3. Set a non-zero angle such as `1.0` and click `Update`
4. Drag a Signal region of interest. Draw a second region of interest with overlapping y. This should cause the reduction to fail and the reduction plot will clear

Part of #34220

*This does not require release notes* because **it is still being developed**

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by sf1919** [comment]
I'm getting quite a few warnings/error messages when I try to test this. When I initially open the ISIS Reflectometry GUI and click on the Preview Tab I get the following warning
```Tight layout not applied. The bottom and top margins cannot be made large enough to accommodate all axes decorations.```

When I load the file ```INTER45455_inst``` I get an error message
Screenshot (222)

When I try to draw a signal box I get another error
Screenshot (225)

And trying to draw overlapping signal boxes (not 100% sure I'm following the instructions correctly) give me this and I'm not sure the plot is clearing as expected.
Screenshot (226)

Do these screenshots look like what you would expect?

**Comment by robertapplin** [comment]
> I'm getting quite a few warnings/error messages when I try to test this. 

Yes those warnings are expected. Apologies the instructions probably aren't very clear. I forgot to say that you need to set the angle to be non-zero, such as 1.0, and then click `Update`, before drawing the regions

And drawing overlapping boxes is expected to show an error, so that is fine

**Comment by robertapplin** [comment]
Thanks for finding that @sf1919 . It should hopefully be working now

File: qt/python/mantidqt/mantidqt/widgets/regionselector/test/test_regionselector_presenter.py [hunk 1]
```diff
@@ -87,6 +87,27 @@ def test_add_second_rectangular_region_deactivates_first_selector(self):
...
+
+        region_selector.update_workspace(mock_ws)
+        region_selector.add_rectangular_region("test", "black")
+        region_selector.add_rectangular_region("test", "black")
+
+        region_selector.clear_workspace()
+
+        self.assertEqual(0, len(region_selector._selectors))
+        self.assertEqual(None, region_selector.model.ws)
+
```
Comment (sf1919) [inline qt/python/mantidqt/mantidqt/widgets/regionselector/test/test_regionselector_presenter.py:102]: Does this mean that in addition to the plot clearing when two overlapping signal areas are drawn that the signal areas should also disappear?

Comment (robertapplin) [reply qt/python/mantidqt/mantidqt/widgets/regionselector/test/test_regionselector_presenter.py:102]: Not quite. This is for when there is an error when summing banks, then the 2nd plot (with the selectors on it) should be cleared. I don't know of a way to test this unfortunately without making a code change to make the SumBanks algorithm break

**Review by sf1919 — APPROVED** [review]
All is now working as expected at this stage. I assume documentation of this new preview tab will come later when it's ready to be deployed, along with the release note.

Code looks fine. Happy to approve.

**Pull request merged**